### ISSUE TITLE
No includePath is used if a custom browse configuration is received and no c_cpp_properties.json exists

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode-cpptools#10773]
With no c_cpp_properties.json, enable a configuration provider so it sends a custom browse config, but then either disable the config provider extension or have it not configured such that no actual custom configuration is sent for a source file.

Bug: No default includePath gets used (defines get used) -- ideally it would fall back to using the recursive includes `${workspaceFolder}/**`. This becomes fixed once the custom browse config is no longer received.

**Comment by sean-mcmanus** [comment]
It seems like this may be fixed with our pending 1.15.2, except for a case where a file is open on workspace load and there's some issue with the database not being ready and so the recursive includes might not be available (which might be a known limitation of recursive includes).

**Comment by Colengms** [comment]
This looks like an issue with recursive includes and some sort of race condition at startup, perhaps racing with the browse database scan.  

I added logging before the recursive includes logic, and I see the workspace folder in the list of paths.  Logging after the recursive includes logic does not include anything under that in the returned results.  If I close the file, wait for the TU to clear out, then open it again, I get the missing paths (from the same scenario).

**Comment by sean-mcmanus** [comment]
@Colengms Yeah, you're describing the behavior after your fixes for 1.15.2 -- with 1.15.1 I'm seeing it never give includePaths -- I can retry with the 1.15.2 vsix after we build it to see if there's some other repro difference too.
